nasa/earthdata-search
# EDSC-3795: Added the customization form for spatial subsetting.

# Overview

### What is the feature?
Added the customization form for spatial subsetting. Gives the option to include the data in the retrieval request. Also displays the coordinates that will be included within the form unless it's a polygon, in which case it'll show the number of points selected.

### What is the Solution?
Added the new form to the AccessMethods as well as the logic to include the data if check and not to include it if unchecked. This logic was added in AccessMethods but several fields were added to constructOrderPayload, buildAccessMethods, and retrievals in order for the params in the outgoing request to be updated correctly. the spatial area 
Summarize what you changed.

### What areas of the application does this impact?

static/src/js/components/AccessMethod/AccessMethod.js (/projects and under customization options)
static/src/js/util/accessMethods/buildAccessMethods.js
static/src/js/util/retrievals.js
serverless/src/submitHarmonyOrder/constructOrderPayload.js

# Testing

### Reproduction steps

- **Environment for testing: Local**
- **Collection to test with: Any Collection**

1. navigate to <environment host>/search
2. Check the _Customizable_ option
3. Add a spatial object (rectangle, point, circle, or polygon) to the map.
4. Select any collection (I used _GHRSST Level 4 MUR Global Foundation Sea Surface Temperature Analysis_)
5. Add 3 or 4 granules to your project and click Download
6. After everything loads, click the first '**Customize** Harmony' option and scroll to the bottom
7. Observe that the second form is the Spatial Subsetting form.
8. Download the Data.
9. Repeat steps 1-7
10. When getting to the Spatial Subsetting form, uncheck the checkbox and download.
11. Navigate to the <environment host>/admin/retrievals and open up the 2 retrievals. Ensure (using the network tab in developer tools) that the payloads matched the request sent (regarding enableSpatialSubsetting)

### Attachments
<img width="1722" alt="image" src="https://github.com/nasa/earthdata-search/assets/8591291/ed6dc77b-38d2-46a3-9c6a-d82a06113ed5">
<img width="1722" alt="image" src="https://github.com/nasa/earthdata-search/assets/8591291/cca8c627-3428-4c33-b066-d4f1e29006b8">
<img width="1722" alt="image" src="https://github.com/nasa/earthdata-search/assets/8591291/6b452ae9-2f07-471a-8eb7-bc20bbf10eff">

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist
